File: benchmarks/benchmarks/05_ml_primitives/comparisons/tensor_matmul.py
def matmul_score(size: int) -> int:
    total = 0
    for row in range(size):
        for col in range(size):
            cell = 0
            for inner in range(size):
                cell += ((row + inner) % 31) * ((inner + col) % 29)
            total += cell
    return total


if __name__ == "__main__":
    print(matmul_score(48))

File: benchmarks/benchmarks/05_ml_primitives/comparisons/tensor_matmul.rs
fn matmul_score(size: i64) -> i64 {
    let mut total = 0;
    for row in 0..size {
        for col in 0..size {
            let mut cell = 0;
            for inner in 0..size {
                cell += ((row + inner) % 31) * ((inner + col) % 29);
            }
            total += cell;
        }
    }
    total
}

fn main() {
    println!("{}", matmul_score(48));
}

